src/components/tools/ManipulatorTool.vue

```diff
@@ -38,38 +39,98 @@ export default defineComponent({
         : viewProxy.value.getInteractorStyle3D();
     });
 
-    const manipulator = props.manipulatorClass.newInstance(options.value);
+    const manipulators: any[] = [];
+    options.value.forEach((opt) =>
+      manipulators.push(props.manipulatorClass.newInstance(opt))
+    );
+
+    function updateStyle(
+      style: any,
+      newManipulators: any[],
+      oldManipulators: any[]
+    ) {
+      if (style !== null) {
+        oldManipulators.forEach((m) => {
+          if (m.isA('vtkCompositeMouseManipulator')) {
+            style.removeMouseManipulator(m);
+          }
+        });
+
+        newManipulators.forEach((m) => {
+          if (m.isA('vtkCompositeMouseManipulator')) {
+            style.addMouseManipulator(m);
+          }
+        });
+      }
+    }
 
+    /*
     watch(options, (newOptions) => {
```

Comment:
This commented function is an alternative approach. It will delete/create all mouse manipulators rather than updating existing ones and only inserting/deleting when needed.

Reply:
This approach does have a (small) perf overhead, due to the deletion/creation of more objects. While this approach is cleaner, I think we can keep the current impl.